packages/ketcher-macromolecules/src/components/monomerLibrary/RnaBuilder/RnaAccordion/RnaAccordion.tsx

```diff
@@ -91,11 +94,48 @@ export const RnaAccordion = ({ libraryName, duplicatePreset, editPreset }) => {
 
   const [expandedAccordion, setExpandedAccordion] =
     useState<RnaBuilderItem | null>(activeRnaBuilderItem);
+  const [newPreset, setNewPreset] = useState(activePreset);
 
+  useEffect(() => {
+    setNewPreset(activePreset);
+  }, [activePreset]);
+
+  const dispatch = useDispatch();
   const handleAccordionSummaryClick = (rnaBuilderItem: RnaBuilderItem) => {
     if (expandedAccordion === rnaBuilderItem) {
       setExpandedAccordion(null);
-    } else setExpandedAccordion(rnaBuilderItem);
+    } else {
+      dispatch(setActiveRnaBuilderItem(rnaBuilderItem));
```

Comment:
Also I'm not quite sure about how to improve reusability here if it's really needed to use this logic, doing so React way would probably cause to restructure components composition to handle common logic
I'd propose to introduce some util function to encapsulate this logic there and call it here and in another component

Reply:
Moved this to helper functions section